storage/fanout_test.go

```diff
@@ -195,10 +195,18 @@ func TestMergeQuerierWithChainMerger(t *testing.T) {
 			}
 			qs = append(qs, tc.extraQueriers...)
 
-			merged, _, _ := NewMergeQuerier(qs[0], qs, ChainedSeriesMerge).Select(false, nil)
-			for merged.Next() {
+			mergedQuerier, _, _ := NewMergeQuerier(qs[0], qs, ChainedSeriesMerge).Select(false, nil)
+
+			// Get all merged series upfront to make sure there are no incorrectly retained shared
+			// buffers causing bugs.
+			var mergedSeries []Series
+			for mergedQuerier.Next() {
+				mergedSeries = append(mergedSeries, mergedQuerier.At())
```

Comment:
Eh, again we are hit but "broken" usage of PromQL. Hopefully, we can fix it here: https://github.com/prometheus/prometheus/issues/7326 

This is good hot fix for now :muscle: